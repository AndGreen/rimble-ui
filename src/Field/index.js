import React from 'react'
import styled from 'styled-components'

import Box from '../Box'
import Text from '../Text'

const RequiredLabel = styled(Text)`
  &:after {
    content: ' (required)'
  }
`

const Field = ({ children, label, ...props }) => {
  console.log(children.props.required)
  return (
    <Box
      as='label'
      display='flex'
      flexDirection='column'
      alignItems='flex-start'
      mb={3}
      {...props}
    >

      {!children.props.required && <Text fontSize={1} fontWeight={3} mb={2}>{label}</Text> }
      {children.props.required && <RequiredLabel fontSize={1} fontWeight={3} mb={2}>{label}</RequiredLabel> }
      {children}
    </Box>
  )
}

Field.displayName = 'Field'

export default Field
