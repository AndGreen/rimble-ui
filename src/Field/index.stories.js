import React from 'react'

import { storiesOf } from '@storybook/react'

import Field from '../Field'
import Input from '../Input'

storiesOf('Field', module)
  .add('text', () => (
    <Field label='label'>
      <Input type='text' />
    </Field>
  ))
