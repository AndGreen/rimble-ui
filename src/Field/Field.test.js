import Field from './'

describe('Field component sanity', () => {
  it('has name', () => {
    expect(Field.displayName).toBe('Field')
  })

  it('matches default snapshot', () => {
    const component = render(<Field />)
    expect(component).toMatchSnapshot()
  })
})
