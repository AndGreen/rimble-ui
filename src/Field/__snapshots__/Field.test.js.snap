// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component sanity matches default snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-bottom: 8px;
  color: #3F3D4B;
  font-size: 14px;
  text-align: left;
  font-size: 14px;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-weight: 600;
  text-align: left;
  line-height: 1.5;
}

.c1 {
  box-sizing: border-box;
}

<label
  class="c0"
  display="flex"
>
  <div
    class="c1"
    color="copyColor"
    font-family="sansSerif"
    font-size="1"
    font-weight="3"
  >
    
  </div>
</label>
`;
